sootup.callgraph/src/main/java/sootup/callgraph/AbstractCallGraphAlgorithm.java

```diff
@@ -133,20 +180,122 @@ final void processWorkList(
+              if (stmt.containsFieldRef() && stmt.getFieldRef() instanceof JStaticFieldRef) {
+                targetsToStaticInitializer.add(
+                    stmt.getFieldRef().getFieldSignature().getDeclClassType());
+              }
+
+              // constructor calls
+              if (stmt instanceof JAssignStmt) {
+                Value rightOp = ((JAssignStmt<?, ?>) stmt).getRightOp();
+                ClassType classType = null;
+                if (rightOp instanceof JNewExpr) {
+                  Type type = rightOp.getType();
+                  if (type instanceof ClassType) {
+                    classType = (ClassType) type;
+                  }
+                } else if (rightOp instanceof JNewArrayExpr) {
+                  classType = findClassTypeInType(((JNewArrayExpr) rightOp).getBaseType());
+                } else if (rightOp instanceof JNewMultiArrayExpr) {
+                  classType = findClassTypeInType(((JNewMultiArrayExpr) rightOp).getBaseType());
+                }
+                if (classType != null) {
+                  targetsToStaticInitializer.add(classType);
+                }
+              }
+
+              // static method calls
+              if (stmt.containsInvokeExpr() && stmt.getInvokeExpr() instanceof JStaticInvokeExpr) {
+                targetsToStaticInitializer.add(
+                    stmt.getInvokeExpr().getMethodSignature().getDeclClassType());
+              }
+            });
+
+    return targetsToStaticInitializer
+        .build()
+        .flatMap(
+            classType ->
+                Stream.concat(
+                    Stream.of(classType), typeHierarchy.superClassesOf(classType).stream()))
+        .filter(Objects::nonNull)
+        .map(classType -> findClinitMethodOfClassType(view, classType))
+        .filter(Optional::isPresent)
+        .map(Optional::get)
+        .map(SootClassMember::getSignature);
+  }
+
+  private ClassType findClassTypeInType(Type type) {
+    if (type instanceof ArrayType) {
+      return findClassTypeInType(((ArrayType) type).getBaseType());
+    }
+    if (type instanceof ClassType) {
+      return (ClassType) type;
     }
+    return null;
+  }
+
+  /**
+   * This methods returns the SootMethod of a clinit method if the requested class contains an
+   * implementation of the clinit method.
+   *
+   * @param view contains the method data
+   * @param classType the type of the class that might contain the clinit method
+   * @return an optional of a SootMethod. The SootMethod is contained if the class described by the
+   *     class type contains an implementation of clinit, otherwise it is empty
+   */
+  private Optional<? extends SootMethod> findClinitMethodOfClassType(
+      @Nonnull View<? extends SootClass<?>> view, @Nonnull ClassType classType) {
+    MethodSignature clinitSig =
+        new MethodSignature(
+            classType,
+            new MethodSubSignature("<clinit>", Collections.emptyList(), VoidType.getInstance()));
```

Comment:
can we refactor that into ClassType.createClinitSignature() ?

Reply:
mmh I will create a getStaticInitializer() in ClassType.

Reply:
I have removed the method and replaced it in its usages with classtype.getStaticInitializer